### PR TITLE
+ kamon-elasticsearch: new integration module

### PR DESCRIPTION
**Pull request by jtjeferreira**
This is a new module for elasticsearch integration, inspired on the jdbc module

### CODE REVIEW

File: kamon-elasticsearch/src/test/scala/kamon/elasticsearch/instrumentation/RequestInstrumentationSpec.scala [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+      val segmentSnapshot = takeSnapshotOf("Elasticsearch[DeleteRequest]", "trace-segment",
+        tags = Map(
+          "trace" -> "elasticsearch-trace-delete",
+          "category" -> SegmentCategory.Database,
+          "library" -> ElasticsearchExtension.SegmentLibraryName))
+
+      segmentSnapshot.histogram("elapsed-time").get.numberOfMeasurements should be(100)
+
+    }
+    //
```
Comment (dpsoft) [inline kamon-elasticsearch/src/test/scala/kamon/elasticsearch/instrumentation/RequestInstrumentationSpec.scala]: @jtjeferreira please remove commented code.

File: kamon-elasticsearch/src/test/scala/kamon/elasticsearch/instrumentation/RequestInstrumentationSpec.scala [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        for (id ← 1 to 10) {
+          intercept[ExecutionException] {
+            client.delete(new delete.DeleteRequest("index name with spaces", "tweet", id.toString)).get()
+          }
+        }
+
+        Tracer.currentContext.finish()
+      }
+
+      val jdbcSnapshot = takeSnapshotOf("elasticsearch-requests", "elasticsearch-requests")
```
Comment (dpsoft) [inline kamon-elasticsearch/src/test/scala/kamon/elasticsearch/instrumentation/RequestInstrumentationSpec.scala]: @jtjeferreira please rename jdbc in variables names.

File: kamon-elasticsearch/src/test/scala/kamon/elasticsearch/instrumentation/AsyncRequestInstrumentationSpec.scala [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+        tags = Map(
+          "trace" -> "elasticsearch-trace-delete",
+          "category" -> SegmentCategory.Database,
+          "library" -> ElasticsearchExtension.SegmentLibraryName))
+
+      segmentSnapshot.histogram("elapsed-time").get.numberOfMeasurements should be(100)
+
+    }
+    //
+    //    "record the execution time of SLOW QUERIES based on the kamon.jdbc.slow-query-threshold" in {
```
Comment (dpsoft) [inline kamon-elasticsearch/src/test/scala/kamon/elasticsearch/instrumentation/AsyncRequestInstrumentationSpec.scala]: @jtjeferreira please remove commented code.

**Comment by dpsoft** [comment]
@jtjeferreira the PR looks very good, please review the comments and I would like to merge it BUT there is one missing piece, not in this repository but in our [documentation website]! yes, I know, it is a bit boring to write docs but it is a necessary thing... so if you could create a simple one page doc explaining what this module does, how it reports data and possibly some screenshots of what kind of visualization or results you can get out of it would be super helpful even for us, to start experimenting with this :D.. take a look at the Datadog, StatsD and New Relic pages on our website to get an idea of what we would like to see there.

BTW: @ivantopo WDYT

[documentation website]: https://github.com/kamon-io/kamon.io

**Comment by jtjeferreira** [comment]
Hi. Sorry I missed these notifications somehow. I will fix the issues you mentioned and work in documentation...

**Comment by jtjeferreira** [comment]
Hi @dpsoft and @ivantopo. I made the changes you requested.

Regarding the documentation, I am working on it although I have some questions. You serve it with github-pages and jekyll? I used the [gem github-pages](https://github.com/github/pages-gem) but if fails in the sitemap_generator.rb

**Comment by dpsoft** [comment]
hi @jtjeferreira you should run ```jekyll serve --watch```, and yes the ```sitemap-generator plugin``` have some issues... maybe disabling the plugin for a while can help you :(

**Comment by jtjeferreira** [comment]
i have fixed the sitemap with https://github.com/jtjeferreira/kamon.io/commit/fb0dfac661d80a50d2ee888ed485479c83c20997 and it works. I am now trying to write docs for both elastic and jdbc since they are very similar

**Comment by dpsoft** [comment]
@jtjeferreira Amazing!!! and please include your fix for Jekyll :)

**Comment by jtjeferreira** [comment]
Hi. Finally managed to write the documentation https://github.com/kamon-io/kamon.io/pull/16

**Comment by dpsoft** [comment]
@jtjeferreira thanks for this PR!!!

**Pull request merged**